src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/ReactiveCourseRepositoryIT.java

```diff
@@ -0,0 +1,197 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Sort;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import javax.annotation.PreDestroy;
+import java.util.Arrays;
+import java.util.Collections;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = TestRepositoryConfig.class)
+public class ReactiveCourseRepositoryIT {
+
+    private static final String COURSE_ID_1 = "1";
+    private static final String COURSE_ID_2 = "2";
+    private static final String COURSE_ID_3 = "3";
+    private static final String COURSE_ID_4 = "4";
+    private static final String COURSE_ID_5 = "5";
+
+    private static final String COURSE_NAME_1 = "Course1";
+    private static final String COURSE_NAME_2 = "Course2";
+    private static final String COURSE_NAME_3 = "Course3";
+    private static final String COURSE_NAME_4 = "Course4";
+    private static final String COURSE_NAME_5 = "Course5";
+
+    private static final String DEPARTMENT_NAME_1 = "Department1";
+    private static final String DEPARTMENT_NAME_2 = "Department2";
+    private static final String DEPARTMENT_NAME_3 = "Department3";
+
+    private static final Course COURSE_1 = new Course(COURSE_ID_1, COURSE_NAME_1, DEPARTMENT_NAME_3);
+    private static final Course COURSE_2 = new Course(COURSE_ID_2, COURSE_NAME_2, DEPARTMENT_NAME_2);
+    private static final Course COURSE_3 = new Course(COURSE_ID_3, COURSE_NAME_3, DEPARTMENT_NAME_2);
+    private static final Course COURSE_4 = new Course(COURSE_ID_4, COURSE_NAME_4, DEPARTMENT_NAME_1);
+    private static final Course COURSE_5 = new Course(COURSE_ID_5, COURSE_NAME_5, DEPARTMENT_NAME_1);
+
+    private final CosmosEntityInformation<Course, String> entityInformation =
+        new CosmosEntityInformation<>(Course.class);
+
+    @Autowired
+    private ReactiveCosmosTemplate template;
+
+    @Autowired
+    private ReactiveCourseRepository repository;
+
+    @PreDestroy
+    public void cleanUpCollection() {
+        template.deleteContainer(entityInformation.getCollectionName());
+    }
+
+    @Before
+    public void setup() {
+        final Flux<Course> savedFlux = repository.saveAll(Arrays.asList(COURSE_1, COURSE_2,
+            COURSE_3, COURSE_4));
+        StepVerifier.create(savedFlux).thenConsumeWhile(course -> true).expectComplete().verify();
+    }
+
+    @After
+    public void cleanup() {
+        final Mono<Void> deletedMono = repository.deleteAll();
+        StepVerifier.create(deletedMono).thenAwait().verifyComplete();
+    }
+
+    @Test
+    public void testFindById() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/Microsoft/spring-data-cosmosdb/pullRequest?prid=4741022)